cli/cli_app.go

```diff
@@ -326,15 +326,25 @@ func runTerragruntWithConfig(terragruntOptions *options.TerragruntOptions, terra
 	}
 
 	beforeHookErrors := processHooks(terragruntConfig.Terraform.GetBeforeHooks(), terragruntOptions)
-	terraformError := runTerraformCommandIfNoErrors(beforeHookErrors, terragruntOptions)
+	terraformError := runTerraformCommandIfNoErrors(beforeHookErrors, terragruntOptions, terragruntConfig)
 	postHookErrors := processHooks(terragruntConfig.Terraform.GetAfterHooks(), terragruntOptions, beforeHookErrors, terraformError)
 
 	return errors.NewMultiError(beforeHookErrors, terraformError, postHookErrors)
 }
 
-func runTerraformCommandIfNoErrors(possibleErrors error, terragruntOptions *options.TerragruntOptions) error {
+func runTerraformCommandIfNoErrors(possibleErrors error, terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
 	if possibleErrors == nil {
-		return shell.RunTerraformCommand(terragruntOptions, terragruntOptions.TerraformCliArgs...)
+		output, err := shell.RunTerraformCommandAndCaptureOutput(terragruntOptions, terragruntOptions.TerraformCliArgs...)
+
+		// show the output we captured
+		terragruntOptions.Logger.Println(output)
```

Comment:
Does no output show until this point? If so, we'll want to actually stream the output to `stdout` / `stderr` *and* capture it. Otherwise, for a long-running command (e.g., RDS), it may be 20 min with no output. See here for how to do both: https://github.com/gruntwork-io/terratest/blob/master/modules/shell/command.go#L51

Reply:
yes, that is one of the concerns I had with changing the `runTerraformCommandIfNoErrors` function (called from`runTerragruntWithConfig`) to use the `CaptureOutput` command. The streaming output from the underlying terraform execution is lost until we replay it.

Looking at the `terratest` implementation, I'll have to figure out to incorporate that with the `RunShellCommand` which is what the `CaptureOutput` command ultimately uses, while opening a buffer for stdOut

Debating yet another `runTerraformCommand....` function, or modifying the `CaptureOutput` to stream the Pipe.